test/MinimalDelegation.execute.t.sol

```diff
@@ -750,4 +744,58 @@ contract MinimalDelegationExecuteTest is TokenHandler, DelegationHandler, Execut
         // Verify the nonce was not incremented
         assertEq(signerAccount.getSeq(0), 0);
     }
+
+    /**
+     * Edge case tests
+     */
+    function test_execute_batch_emptyCalls_succeeds() public {
+        Call[] memory calls = CallUtils.initArray();
+        vm.prank(address(signerAccount));
+        signerAccount.execute(BATCHED_CALL, abi.encode(calls));
+    }
+
+    function test_execute_batch_emptyCalls_revertsWhenUnauthorized() public {
+        Call[] memory calls = CallUtils.initArray();
+        vm.expectRevert(IERC7821.Unauthorized.selector);
+        signerAccount.execute(BATCHED_CALL, abi.encode(calls));
+    }
+
+    /**
+     * Self call tests
+     */
+    function test_execute_register_update_asRoot_succeeds() public {
+        TestKey memory newKey = TestKeyManager.initDefault(KeyType.Secp256k1);
+        Call[] memory calls = CallUtils.initArray();
+        calls = calls.push(CallUtils.encodeRegisterCall(newKey));
+        calls = calls.push(CallUtils.encodeUpdateCall(newKey.toKeyHash(), Settings.wrap(0)));
+
+        vm.prank(address(signerAccount));
+        signerAccount.execute(BATCHED_CALL, abi.encode(calls));
+    }
+
+    function test_execute_register_update_asNonRoot_reverts() public {
+        TestKey memory newKey = TestKeyManager.initDefault(KeyType.Secp256k1);
+        Call[] memory calls = CallUtils.initArray();
+        calls = calls.push(CallUtils.encodeRegisterCall(newKey));
+        calls = calls.push(CallUtils.encodeUpdateCall(newKey.toKeyHash(), Settings.wrap(0)));
```

Comment:
this would be interesting if we sent these in with the mode BATCHED_CALL_CAN_REVERT and expect no state to change but no revert to happen

Reply:
Ooo good idea

Reply:
Ah so the Unauthorized is thrown during validation not execution so the mode isn't applied